src/app/client/src/app/modules/learn/components/learn-page/learn-page.component.ts

```diff
@@ -298,7 +298,28 @@ export class LearnPageComponent implements OnInit, OnDestroy, AfterViewInit {
   }
 
   public viewAll(event) {
-    const searchQuery = JSON.parse(event.searchQuery);
+    const userData = this.userService.userProfile;
+    const requestSearchQuery = JSON.stringify({
+      'request': {
+        'filters': {
+          'contentType': [
+            'Course'
+          ],
+          'objectType': [
+            'Content'
+          ],
+          'status': [
+            'Live'
+          ]
+        },
+        'sort_by': {
+          'lastPublishedOn': 'desc'
+        },
+        'limit': 10,
+        'organisationId': _.get(userData, 'organisationIds')
+      }
+    });
+    const searchQuery = _.get(event, 'searchQuery') ?  JSON.parse(event.searchQuery) : JSON.parse(requestSearchQuery);
```

Comment:
requestSearchQuery is default here. It should not.

Only on click of "view All" button of myCourses, we should use requestSearchQuery.

Reply:
it will be that way only. if the event has "searchQuery", then we will use this. else our given searchQuery will be used.